TWTValidation/JSON Validator/Processors/TWTJSONObjectValidatorGenerator.m

```diff
@@ -0,0 +1,410 @@
+    }
+
+    [self pushJSONObjectValidatorWithCommonValidator:commonValidator typeValidator:typeValidator node:genericNode];
+}
+
+
+- (void)processArrayNode:(TWTJSONSchemaArrayASTNode *)arrayNode
+{
+    TWTValidator *commonValidator = [self commonValidatorFromNode:arrayNode];
+
+    TWTValidator *itemValidator = nil;
+    NSArray *indexedItemValidators = nil;
+    if (arrayNode.itemSchema) {
+        itemValidator = [self validatorFromNode:arrayNode.itemSchema];
+    } else {
+        indexedItemValidators = [self validatorsFromNodeArray:arrayNode.indexedItemSchemas];
+    }
+
+    TWTValidator *additionalItemsValidator = [self validatorFromNode:arrayNode.additionalItemsNode];
+    TWTJSONSchemaArrayValidator *typeValidator = [[TWTJSONSchemaArrayValidator alloc] initWithMaximumItemCount:arrayNode.maximumItemCount
+                                                                                              minimumItemCount:arrayNode.minimumItemCount
+                                                                                           requiresUniqueItems:arrayNode.requiresUniqueItems
+                                                                                                 itemValidator:itemValidator
+                                                                                         indexedItemValidators:indexedItemValidators
+                                                                                      additionalItemsValidator:additionalItemsValidator];
+
+    [self pushJSONObjectValidatorWithCommonValidator:commonValidator typeValidator:typeValidator node:arrayNode];
+}
+
+
+- (void)processNumberNode:(TWTJSONSchemaNumberASTNode *)numberNode
+{
+    TWTValidator *commonValidator = [self commonValidatorFromNode:numberNode];
+
+    [self pushNewObject:[[NSMutableArray alloc] init]];
+
+    [self addSubvalidator:[[TWTBlockValidator alloc] initWithBlock:^BOOL(id value, NSError *__autoreleasing *outError) {
+        return [value isKindOfClass:[NSNumber class]] && strcmp([(NSValue *)value objCType], @encode(BOOL)) != 0;
+    }]];
+
+    if (numberNode.minimum || numberNode.maximum || numberNode.requireIntegralValue) {
+        TWTNumberValidator *validator = [[TWTNumberValidator alloc] initWithMinimum:numberNode.minimum maximum:numberNode.maximum];
+        validator.maximumExclusive = numberNode.exclusiveMaximum;
+        validator.minimumExclusive = numberNode.exclusiveMinimum;
+        validator.requiresIntegralValue = numberNode.requireIntegralValue;
+        [self addSubvalidator:validator];
+    }
+
+    if (numberNode.multipleOf) {
+        double multipleOfValue = numberNode.multipleOf.doubleValue;
+        [self addSubvalidator:[[TWTBlockValidator alloc] initWithBlock:^BOOL(id value, NSError *__autoreleasing *outError) {
+            double result = [(NSNumber *)value doubleValue] / multipleOfValue;
+            return result == trunc(result);
```

Comment:
Two things: 

1. You don’t know value is an NSNumber, do you?
2. You can do this a little more cleanly using `fmod`. 

```objc
return fmod([value doubleValue], multipleOfValue]) == 0.0;
```